src/containers/pages/OrganizationViews/AssignPractitioners/tests/index.test.tsx

```diff
@@ -216,6 +216,59 @@ describe('src/pages/*/AssignPractitioners', () => {
     expect(wrapper.find('Select').props().value).toEqual([]);
   });
 
+  it('Should not navigate when there are unsaved changes', async () => {
+    fetch
+      .once(JSON.stringify(fixtures.organization3))
+      .once(JSON.stringify([]))
+      .once(JSON.stringify(fixtures.allPractitioners));
+
+    const mock: any = jest.fn();
+    const props = {
+      assignedPractitioners: [],
+      fetchOrganizationsCreator: fetchOrganizations,
+      history,
+      location: mock,
+      match: {
+        isExact: true,
+        params: { id: fixtures.organization3.identifier },
+        path: `${ASSIGN_PRACTITIONERS_URL}/:id`,
+        url: `${ASSIGN_PRACTITIONERS_URL}/${fixtures.organization3.identifier}`,
+      },
+      organization: fixtures.organization3,
+      serviceClass: OpenSRPService,
+    };
+
+    const wrapper = mount(
+      <Router history={history}>
+        <AssignPractitioner {...props} />
+      </Router>
+    );
+
+    await flushPromises();
+    wrapper.update();
+
+    const select = wrapper.find('Select');
+    // simulate single value change function
+    const entry = fixtures.practitioner2;
+    (select.instance() as any).selectOption({
+      label: `${entry.username} - ${entry.name}`,
+      value: entry.identifier,
+    });
+    wrapper.update();
+
+    // try navigating with usaved changes
+    history.push(FI_URL);
+    let locIsLogin = location.pathname === FI_URL;
+    expect(locIsLogin).toBeFalsy();
```

Comment:
@ciremusyoka at this point we should have an alert pop up, right?  It should be possible to test that this happened, and that the alert contained the correct message.

Reply:
Agreed, this far it doesn't look like these tests are testing for the alert.